### PR TITLE
feat(update-user): update user fields

### PR DESCRIPTION
**Pull request by enigbe**
### What does this PR do?
- Adds functionality to get an existing user by user ID
- Adds functionality to update user's fields
### Tasks completed
- Find user by ID: `GET '/users/{id}'` where `id` corresponds to a MongoDB document id (`_id`)
- Update user of a given ID: `PATCH '/users/{id}'`
### How should this be tested?
With postman, take the following steps:
- Start the server: `go run main.go`
- Create a user: `POST '/users'`. Provide a body with these JSON fields `{"FirstName": "Jane", "LastName": "Doe", "Email": "janedoe@email.com", "password": "complexpassword"}`
- Read created user: `GET '/users/{id}'` where `id` is contained in the response from creating a user. 
- Update some user field: `PATCH '/users/{id}'`. Provide a body `{
    "FirstName": "Millicent",
    "LastName": "Bystander",
    "Phone": "987654321",
    "Company": "Zuri Chat",
    "Organizations": ["Zuri Main", "Zuri Core"]
}` and then read the user to check the updated fields
### Pending Tasks
- User input validation
### Issue Reference
- #37 
- #99 

### Status: Work-In-Progress
- A rudimentary implementation of user field updates without validation
- Endpoints are not authenticated. Awaiting authentication from: 
     - #155 
     - #113 
     - #118

### CODE REVIEW

File: user/user.go [hunk 1]
```diff
@@ -51,17 +54,58 @@ func Create(response http.ResponseWriter, request *http.Request) {
...
-// 	}
-// 	return user, nil
-// }
+func FindUserByID(response http.ResponseWriter, request *http.Request) {
+	// Find a user by user ID
+	response.Header().Set("content-type", "application/json")
+
+	collectionName := "users"
+	userID := mux.Vars(request)["id"]
+	objID, _ := primitive.ObjectIDFromHex(userID)
```
Comment (Aminujb) [inline user/user.go:63]: Please cater for a situation where by the supplied Id is a wrong one and primitive throws error. Currently with this, the application will get stuck when an invalid primitive is supplied.

File: user/user.go [hunk 2]
```diff
@@ -51,17 +54,58 @@ func Create(response http.ResponseWriter, request *http.Request) {
...
+
+}
+
+func UpdateUser(response http.ResponseWriter, request *http.Request) {
+	// Update a user of a given ID
+	response.Header().Set("content-type", "application/json")
+
+	collectionName := "users"
+	userID := mux.Vars(request)["id"]
+	objID, _ := primitive.ObjectIDFromHex(userID)
```
Comment (Aminujb) [inline user/user.go:80]: Cater for err for invalid primitive

File: user/user.go [hunk 3]
```diff
@@ -51,17 +54,58 @@ func Create(response http.ResponseWriter, request *http.Request) {
...
+
+	res, err := utils.GetMongoDbDoc(collectionName, bson.M{"_id": objID})
+	if err != nil {
+		utils.GetError(err, http.StatusInternalServerError, response)
+		return
+	}
+
+	if res != nil {
+		// 2. Get user fields to be updated from request body
+		var body map[string]interface{}
```
Comment (Aminujb) [inline user/user.go:90]: Please confirm that this doesn't replace the user document with any random object. Eg test with a Json that has nothing to do with a user schema e.g {"school" :"unilag"}

**Review by Aminujb — CHANGES_REQUESTED** [review]
Please attend to the raised reviews. Nice work done.

File: user/user.go (hunk 1 above)
Comment (enigbe) [reply user/user.go:63]: Resolved this a few minutes ago.

File: user/user.go (hunk 2 above)
Comment (enigbe) [reply user/user.go:80]: Resolved also.

File: user/user.go (hunk 3 above)
Comment (enigbe) [reply user/user.go:90]: Currently working on this.

Comment (enigbe) [reply user/user.go]: I have made some updates to this but still have the problem as earlier. Uncertain how to go about this. If there are any pointers you can give I will be grateful.

**Pull request merged**